pkg/operator/operator.go

```diff
@@ -46,6 +46,61 @@ var (
 	setupLog = log.Log.WithName("operator")
 )
 
+func isEssentialAnnotation(key string) bool {
+	// Keep trivy-operator specific annotations
+	if strings.HasPrefix(key, "trivy-operator.") ||
+		strings.HasPrefix(key, "aquasecurity.github.io/") ||
+		strings.HasPrefix(key, "trivyoperator.") {
+		return true
+	}
+
+	// Keep essential Kubernetes annotations
+	// ref: https://kubernetes.io/docs/reference/labels-annotations-taints/
+	essentialAnnotations := []string{
+		"app.kubernetes.io/",
+		"kubernetes.io/",
+		"service.kubernetes.io/",
+		"deployment.kubernetes.io/revision",
+	}
```

Comment:
IIUC `isEssentialAnnotation` is called a lot of time, so maybe should it be constants to avoid memory allocation for every call?
 
also I'd consider to use `map` for some cases like `app` or `version`.